src/gui/qgsdatasourcemanagerdialog.cpp

```diff
@@ -59,12 +60,49 @@ QgsDataSourceManagerDialog::QgsDataSourceManagerDialog( QgsBrowserGuiModel *brow
   mBrowserWidget->setFeatures( QDockWidget::NoDockWidgetFeatures );
   ui->mOptionsStackedWidget->addWidget( mBrowserWidget );
   mPageNames.append( QStringLiteral( "browser" ) );
+
   // Forward all browser signals
   connect( mBrowserWidget, &QgsBrowserDockWidget::handleDropUriList, this, &QgsDataSourceManagerDialog::handleDropUriList );
   connect( mBrowserWidget, &QgsBrowserDockWidget::openFile, this, &QgsDataSourceManagerDialog::openFile );
   connect( mBrowserWidget, &QgsBrowserDockWidget::connectionsChanged, this, &QgsDataSourceManagerDialog::connectionsChanged );
   connect( this, &QgsDataSourceManagerDialog::updateProjectHome, mBrowserWidget->browserWidget(), &QgsBrowserWidget::updateProjectHome );
 
+  // METADATA Add the metadata search widget as the second stacked widget
+  mLayerMetadataSearchWidget = new QgsLayerMetadataSearchWidget( mMapCanvas, this );
+  connect( mLayerMetadataSearchWidget, &QgsLayerMetadataSearchWidget::rejected, this, &QgsDataSourceManagerDialog::reject );
+  ui->mOptionsStackedWidget->addWidget( mLayerMetadataSearchWidget );
+  mPageNames.append( QStringLiteral( "metadata" ) );
+
+  connect( mLayerMetadataSearchWidget, &QgsLayerMetadataSearchWidget::addLayers, this, [ = ]( const QList< QgsLayerMetadataProviderResult > &metadataResults )
```

Comment:
Why is this code here instead of in QgsLayerMetadataSearchWidget ?

Reply:
I created the new classes and now using source select register.